### PR TITLE
chore(legacy): Introduce public version class and drop version methods from OC_Util

### PR DESCRIPTION
**Pull request by juliusknorr**
- Introduce a ServerVersion class with properly typed methods for getting different version parts and strings
- Remove and migrate any usages of OC_Util to the new API

Next step could be to deprecated related OCP\Util methods.

### CODE REVIEW

File: lib/public/Util.php [hunk 1]
```diff
@@ -37,7 +37,7 @@ class Util {
 	 * @since 4.0.0
```
Comment (come-nc) [inline lib/public/Util.php:37]: `@deprecated`

File: lib/public/Util.php [hunk 2]
```diff
@@ -68,7 +68,7 @@ public static function setChannel($channel) {
 	 * @since 8.1.0
```
Comment (come-nc) [inline lib/public/Util.php:68]: same

**Review by come-nc — APPROVED** [review]

File: lib/private/legacy/OC_Defaults.php [hunk 3]
```diff
@@ -1,5 +1,9 @@
 <?php
 
+use OCP\IConfig;
```
Comment (nickvergessen) [inline lib/private/legacy/OC_Defaults.php:3]: should be after license header

File: lib/public/ServerVersion.php [hunk 4]
```diff
@@ -0,0 +1,94 @@
+<?php
```
Comment (nickvergessen) [inline lib/public/ServerVersion.php:1]: declare strict

File: lib/public/ServerVersion.php [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+ * @since 31.0.0
+ */
+class ServerVersion {
+
+	private array $version;
+	private string $versionString;
+	private string $build;
+	private string $channel;
+
+	public function __construct() {
```
Comment (nickvergessen) [inline lib/public/ServerVersion.php:19]: Required since

File: lib/public/ServerVersion.php [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+
+	/**
+	 * @since 31.0.0
+	 */
+	public function getVersionString(): string {
+		return $this->versionString;
+	}
+
+	/**
+	 * @since 31.0.0
```
Comment (nickvergessen) [inline lib/public/ServerVersion.php:69]:
```suggestion
	 * @psalm-return 'beta'|'stable'|'enterprise'|'git'
	 * @since 31.0.0
```

**Review by nickvergessen — DISMISSED** [review]

**Review by susnux — APPROVED** [review]

**Comment by nickvergessen** [comment]
Since this was defacto-standard, please document the new API and mention the removal in the docs

**Pull request merged**